node/lib/util/commit.js

```diff
@@ -606,182 +643,232 @@ exports.commitPaths = co.wrap(function *(repo, status, message) {
+     *
+     * @param {CommitMetaData} other
+     * @return {Boolean}
+     */
+    equivalent(other) {
+        assert.instanceOf(other, CommitMetaData);
+        return this.d_message === other.d_message &&
+            this.d_signature.name() === other.d_signature.name() &&
+            this.d_signature.email() === other.d_signature.email();
+    }
+}
 
-        const newSub = currentSubs[subName];
+exports.CommitMetaData = CommitMetaData;
 
-        // If the sub has been deleted in the index, it's bad.
+/**
+ * Return the meta-data for the specified `commit`.
+ *
+ * @param {NodeGit.Commit} commit
+ * @return {CommitMetaData}
+ */
+exports.getCommitMetaData = function (commit) {
+    assert.instanceOf(commit, NodeGit.Commit);
+    return new CommitMetaData(commit.committer(), commit.message());
+};
 
-        if (null === newSub.index) {
-            deleted.push(subName);
-            return;                                                   // RETURN
-        }
+/**
+ * Return the amend status for the submodule having the specified current
+ * `status` and the optionally specified `old` value in the previous commit.  A
+ * missing `old` indicates that the submodule was added in the commit being
+ * amended.
+ *
+ * @param {RepoStatus.Submodule} status
+ * @param {Submodule|null}       old
+ * @return {Object}
+ * @return {CommitMetaData|null}       return.oldCommit if sub in last commit
+ * @return {RepoStatus.Submodule|null} return.status    null if shouldn't exist
+ */
+exports.getSubmoduleAmendStatus = co.wrap(function *(status,
+                                                     old,
+                                                     getRepo,
+                                                     all) {
+    assert.instanceOf(status, RepoStatus.Submodule);
+    if (null !== old) {
+        assert.instanceOf(old, Submodule);
+    }
+    else {
+        old = null;
+    }
+    assert.isFunction(getRepo);
+    assert.isBoolean(all);
 
-        // If the submodule didn't exist before, it's inherently OK.
+    const index = status.index;
+    const commit = status.commit;
+    const workdir = status.workdir;
 
-        if (undefined === oldSub) {
-            return;                                                   // RETURN
-        }
+    if (null === old && null === index) {
+        // Gone in index and didn't exist before; has no status now.
 
-        // If a submodule has a different commit fail.
+        return {
+            oldCommit: null,
+            status: null,
+        };
+    }
 
-        const relation = newSub.index.relation;
-        if (RepoStatus.Submodule.COMMIT_RELATION.SAME !== relation) {
-            newCommits[subName] = relation;
-            return;                                                   // RETURN
-        }
+    // We'll do an amend of this sub only if:
+    // - it's not gone in the index
+    // - it wasn't removed in the last commit
+    // - it wasn't added in the last commit
```

Comment:
By "do an amend" I mean perform a commit amendment in the sub-repo; I will clarify. What you are asking should work.